packages/conf/src/index.ts

```diff
@@ -136,13 +138,22 @@ export function getConfig({
     ? configExplorer.load(configPath)
     : configExplorer.search(defaultRootDir)
   const userConfig = result ? result.config : {}
+
+  return makeConfig({
+    rootDir: result ? path.dirname(result.filepath) : defaultRootDir,
+    ...userConfig,
+  }, {skipValidation})
+}
```

Comment:
i split `getConfig`into loading part (`getConfig`) which do all cosmiconfig related stuff and processing part (`makeConfig`) which validates, populate defaults, etc 

The latter is makes testing much easier.